llvm/utils/TableGen/AsmMatcherEmitter.cpp

```diff
@@ -608,12 +627,19 @@ struct MatchableInfo {
   void buildInstructionResultOperands();
   void buildAliasResultOperands(bool AliasConstraintsAreChecked);
 
-  /// operator< - Compare two matchables.
-  bool operator<(const MatchableInfo &RHS) const {
+  /// comp - Compare two matchables.
+  bool comp(const MatchableInfo &RHS, const CodeGenTarget &Target) const {
```

Comment:
We could probably use the chance and give it some meaningful name. Something like `shouldBeMatchedBefore()` maybe?